vcf-dumper/vcf-dumper-ws/src/test/java/uk/ac/ebi/eva/vcfdump/server/rest/VcfDumperIntegrationTest.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    /**
+     * Test segments endpoint from {@link HtsgetVcfController#getHtsgetUrls}
+     */
+    @Test
+    public void getHtsgetUrls() {
+        String url = "/v1/variants/PRJEB9799?format=VCF&referenceName=1&species=ecaballus_20&start=3000000&end=3010000";
+        ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertUrls(response, 3000000, 3010000);
+    }
+
+    private void assertUrls(ResponseEntity<String> response, int start, int end) {
+        HtsGetResponse htsGetResponse = getUrlsFromResponse(response);
+        assertEquals("VCF", htsGetResponse.getFormat());
+        List<UrlResponse> urls = htsGetResponse.getUrls();
+        long numberOfHeaderUrls = urls.stream().filter(u -> u.getUrlClass().equals("header")).count();
+        assertEquals(1, numberOfHeaderUrls);
+        long numberOfBodyUrls = urls.stream().filter(u -> u.getUrlClass().equals("body")).count();
+        int expectedNumberOfBlocks = expectedNumberOfBlocks(start, end);
+        assertEquals(expectedNumberOfBlocks, numberOfBodyUrls);
+    }
+
+    private HtsGetResponse getUrlsFromResponse(ResponseEntity<String> response) {
+        Configuration configuration = Configuration.defaultConfiguration()
+                                                   .jsonProvider(new JacksonJsonProvider())
+                                                   .mappingProvider(new JacksonMappingProvider(objectMapper))
+                                                   .addOptions(Option.SUPPRESS_EXCEPTIONS);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        HtsGetResponse htsGetResponse = JsonPath.using(configuration)
+                                                .parse(response.getBody())
+                                                .read("$['htsget']", new TypeRef<HtsGetResponse>() {});
+        return htsGetResponse;
+    }
+
+    private int expectedNumberOfBlocks(int start, int end) {
+        double positions = end - start;
+        double blocks = Math.ceil(positions / BLOCK_SIZE);
+        return (int) blocks;
+    }
+
+    /**
+     * Test segments endpoint from {@link HtsgetVcfController#getHtsgetUrls}
+     * To retrieve the HTSGET urls it is only needed that the study is present in the files collection. It will return
+     * them even if there is not variants collection.
+     */
+    @Test
+    public void getHtsgetUrlsFromNoVariantsDatabase() {
+        String url = "/v1/variants/1?format=VCF&referenceName=1&species=no_variants&start=1000&end=2000";
+        ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertUrls(response, 1000, 2000);
+    }
+
+    /**
+     * Given that 'end' is exclusive, it would result on 'start' being greater than 'end'
+     */
+    @Test
+    public void checkStartAndEndAreEquals() {
```

Comment:
can you add a similar test for the `segments` endpoint to check that it takes an inclusive `end`?